variant_remapping_tools/reads_to_remapped_variants.py

```diff
@@ -85,86 +76,138 @@ def fetch_bases(fasta, contig, length, start):
-                varpos, perf_counter, local_region_size = calculate_variant_position(read, flank_length)
-                if is_local_alignment_perfect(counter, perf_counter, local_region_size):
-                    name = read.query_name
-                    info = name.split('|')
-                    old_ref = info[2]
-                    old_alt = info[3]
-
-                    new_ref = fetch_bases(fasta, read.reference_name, len(old_ref), varpos)
-                    new_ref, new_alt = calculate_new_alleles(old_ref, new_ref, old_alt, read.is_reverse)
-
+    with open(output_file, 'w') as outfile, open(out_failed_file, 'w') as out_failed:
+        for primary_group, secondary_group in group_reads(bam_file_path):
+            counter['total'] += 1
+            if pass_basic_filtering(primary_group, secondary_group, counter, filter_align_with_secondary,
+                                    alignment_score_threshold):
+                left_read, right_read = _order_reads(primary_group)
+                if pass_aligned_filtering(left_read, right_read, counter):
+                    counter['Remapped'] += 1
+                    varpos, new_ref, new_alts = calculate_new_variant_definition(left_read, right_read, fasta)
+                    info = left_read.query_name.split('|')
                     outfile.write(
-                        '%s\t%s\t%s\t%s\t%s\t%s\t%s\t%s\n' % (read.reference_name, varpos, info[4], new_ref, new_alt,
-                                                              info[5], info[6], info[7]))
-
-    print(counter['total'])
-    print(counter['unmapped'])
-    print(counter['primary_poor'])
-    print(counter['gap_small'])
-    print(counter['context_bad'])
-    print(counter['remapped'])
-    print('% of variants rejected for:')
-    print('Unmapped: {:.2%}'.format(counter['unmapped'] / counter['total']))
-    print('Primary alignment too poor: {:.2%}'.format(counter['primary_poor'] / counter['total']))
-    print('Primary and secondary alignments too close: {:.2%}'.format(counter['gap_small'] / counter['total']))
-    print('Local region around variant too poorly aligned: {:.2%}'.format(counter['context_bad'] / counter['total']))
+                        '%s\t%s\t%s\t%s\t%s\t%s\t%s\t%s\n' % (left_read.reference_name, varpos, info[4], new_ref,
+                                                              ','.join(new_alts), info[5], info[6], info[7]))
+                else:
+                    output_failed_alignment(primary_group, out_failed)
+            else:
+                output_failed_alignment(primary_group, out_failed)
+    for metric in ['Too many alignments', 'Flank unmapped', 'Different chromosomes',
+                   'Poor alignment', 'Overlapping alignment', 'Soft-clipped alignments', 'Remapped']:
+        print(f'{counter.get(metric, 0)} ({counter.get(metric, 0)/counter.get("total"):.2%}) variants rejected for {metric}')
```

Comment:
This should be equivalent to the following:
```suggestion
        print(f'{counter[metric]} ({counter[metric]/counter["total"]:.2%}) variants rejected for {metric}')
```